python_modules/dagster/dagster/_cli/asset.py

```diff
@@ -111,7 +112,46 @@ def asset_list_command(**kwargs):
     asset_keys = asset_selection.resolve(repo_def.asset_graph, allow_missing=True)
 
     for asset_key in sorted(asset_keys):
-        print(asset_key.to_user_string())  # noqa: T201
+        print(asset_key.to_user_string())
+
+
+@asset_cli.command(name="report", help="Report asset materialization")
+@python_origin_target_argument
+@click.option("--select", help="Asset selection to target", required=True)
+@click.option("--partition", help="Asset partition to target", required=False)
+@click.option("--all", is_flag=True, help="Report all partitions", default=False)
```

Comment:
Even though it's a little more verbose, I think it would be clearer to name this `--all-partitions`.